app/controllers/api/v1/subjects_controller.rb

```diff
@@ -20,13 +20,25 @@ def index
   end
 
   def queued
-    selector = Subjects::Selector.new(api_user.user, workflow, params)
+    subject_selector = Subjects::Selector.new(api_user.user, params)
+    selected_subject_ids = subject_selector.get_subject_ids
+
+    selected_subject_scope = Subject
```

Comment:
Wondering if the selector should maybe return a Selection object which would have `subject_ids`, `scope`, and `context` methods that the controller would just be able to call. What do you think?

Reply:
OK, happy to leave it at this